server/chat-plugins/rock-paper-scissors.ts

```diff
@@ -0,0 +1,398 @@
+/**
+ * Rock, Paper, Scissors plugin by Mia
+ * @author mia-pi-git
+ */
+import {Utils} from '../../lib/utils';
+
+const TIMEOUT = 10 * 1000;
+const MAX_ROUNDS = 10;
+
+export class RPSPlayer extends Rooms.RoomGamePlayer {
+	user: User;
+	currentChoice: string;
+	name: string;
+	constructor(user: User, game: RPSGame) {
+		super(user, game);
+		this.user = user;
+		this.name = user.name;
+		this.currentChoice = "";
+	}
+}
+
+export class RPSGame extends Rooms.RoomGame {
+	room: Room;
+	rounds: number;
+	points: {[k: string]: number};
+	currentRound: number;
+	playerTable: {[k: string]: RPSPlayer};
+	readonly checkChat = true;
+	roundTimer?: NodeJS.Timeout;
+	players: RPSPlayer[];
+	wins: ({name: string, choice: string} | null)[];
+	constructor(room: Room, rounds = 3) {
+		super(room);
+		this.room = room;
+		this.rounds = rounds;
+		this.points = {};
+		this.currentRound = 0;
+		this.playerTable = {};
+		this.title = 'Rock, Paper, Scissors';
+		this.gameid = 'rockpaperscissors' as ID;
+		this.players = [];
+		this.wins = [];
+
+		this.room.update();
+		this.room.add(`|controlshtml|<center>Waiting for another player to join....</center>`);
+		this.room.add(`|fieldhtml|<center><h2>Waiting to start the game...</h2></center>`);
+	}
+	onJoin(user: User) {
+		if (user.id in this.playerTable) return;
+		if (Object.keys(this.playerTable).length < 2) {
+			this.addPlayer(user);
+		}
+	}
+	/**
+	 * True - Attacker wins. False - Attacker loses. Null - tie.
+	 */
+	checkMatchup(attacker: string, defender: string): boolean | null {
+		if (attacker === defender) return null;
+		switch (attacker) {
+		case 'rock': {
+			if (defender === 'scissors') return true;
+			if (defender === 'paper') return false;
+			break;
+		}
+		case 'scissors': {
+			if (defender === 'paper') return true;
+			if (defender === 'rock') return false;
+			break;
+		}
+		case 'paper': {
+			if (defender === 'rock') return true;
+			if (defender === 'scissors') return false;
+			break;
+		}
+		}
+		if (!attacker && !defender) return null;
+		return !!(attacker && !defender);
+	}
+	sendOptions() {
+		const button = (cmd: string, title: string) => {
+			return `<button class="button" name="send" value="/${cmd}">${title}</button>`;
+		};
+		let buf = `<center><strong>Make your choice, quick!</strong><br />`;
+		for (const item of ['Rock', 'Paper', 'Scissors']) {
+			buf += `${button(`rps choose ${item}`, item)}`;
```

Comment:
```suggestion
			buf += `${button(`choose ${item}`, item)}`;
```

Does `/choose` not work here?

Reply:
Also why not use the icons the teambuilder uses?